### PR TITLE
Implementing the RouteTradeSwap

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

**Review by DimaCantemir — APPROVED** [review]
lgtm

File: provider/ws.go [hunk 1]
```diff
@@ -197,6 +197,16 @@ func (w *WSClient) PostTradeSwap(ctx context.Context, owner, inToken, outToken s
 	return &response, nil
 }
 
+// PostRouteTradeSwap returns a partially signed transaction for submitting a swap request
```
Comment (kdonthi) [inline provider/ws.go]: "Partially signed transaction(s)"

File: provider/grpc.go [hunk 2]
```diff
@@ -200,14 +205,14 @@ func (g *GRPCClient) SubmitTradeSwap(ctx context.Context, owner, inToken, outTok
...
-		return []string{}, err
+		return nil, err
 	}
 
 	var signatures []string
 	for _, tx := range resp.Transactions {
 		signature, err := g.signAndSubmit(ctx, tx, skipPreFlight)
 		if err != nil {
-			return signatures, err
+			return nil, err
```
Comment (kdonthi) [inline provider/grpc.go]: It would be good to keep the signatures here so we can see what swaps have been completed

**Review by kdonthi — APPROVED** [review]

File: provider/grpc.go (hunk 2 above)
Comment (aspin) [reply provider/grpc.go]: +1. maybe good to wrap the error too, to include info about how many transactions were submitted?

**Review by aspin — APPROVED** [review]

File: provider/grpc.go (hunk 2 above)
Comment (tinystarinagalaxy) [reply provider/grpc.go]: we are returning signatures now.

File: provider/ws.go (hunk 1 above)
Comment (tinystarinagalaxy) [reply provider/ws.go]: done

**Pull request merged**